include/CL/opencl.hpp

```diff
@@ -8859,6 +8931,54 @@ typedef CL_API_ENTRY cl_int (CL_API_CALL *PFN_clEnqueueReleaseD3D10ObjectsKHR)(
     {
         return detail::errHandler(::clFinish(object_), __FINISH_ERR);
     }
+
+#if CL_HPP_TARGET_OPENCL_VERSION >= 300
+    cl_int enqueueWaitSemaphoresKHR(
+        const vector<SemaphoreKhr>* sema_objects,
+        const vector<cl_semaphore_payload_khr>* sema_payloads = NULL,
+        const vector<Event>* events_wait_list = NULL,
+        Event *event = NULL) const
+    {
+        cl_event tmp;
+        cl_int err = detail::errHandler(
+            pfn_clEnqueueWaitSemaphoresKHR(
+                object_,
+                (cl_uint)sema_objects->size(),
+                (const cl_semaphore_khr *) &sema_objects->front(),
+                (sema_payloads != NULL && sema_payloads->size() > 0) ? &sema_payloads->front() : NULL,
+                (events_wait_list != NULL) ? (cl_uint) events_wait_list->size() : 0,
+                (events_wait_list != NULL && events_wait_list->size() > 0) ? (cl_event*) &events_wait_list->front() : NULL,
+                (event != NULL) ? &tmp : NULL),
+            __ENQUEUE_WAIT_SEMAPHORE_KHR_ERR);
+        if (event != NULL && err == CL_SUCCESS)
+            *event = tmp;
+        return err;
+    }
+
+    cl_int enqueueSignalSemaphoreKhr(
+        const vector<SemaphoreKhr>* sema_objects,
+        const vector<cl_semaphore_payload_khr>* sema_payloads = NULL,
```

Comment:
Same comments as for `enqueueWaitSemaphoresKHR`:

```suggestion
        const vector<SemaphoreKhr>& sema_objects,
        const vector<cl_semaphore_payload_khr>& sema_payloads = {},
```

Reply:
Done, thank you!